integrated_channels/integrated_channel/exporters/course_metadata.py

```diff
@@ -40,15 +40,15 @@ def __init__(self, user, enterprise_configuration):
         self.courses = []
         client = EnterpriseApiClient(self.user)
         enterprise_course_runs = client.get_enterprise_course_runs(self.enterprise_customer)
-        LOGGER.info('Retrieving course run list for enterprise %s', self.enterprise_customer.name)
+        LOGGER.info('Retrieving course run list for enterprise [%s]', self.enterprise_customer.name)
         for course_run in enterprise_course_runs.values():
             self.add_course_run(course_run)
 
     def export(self):
         """
-        Export the set of courses generically by encoding them in JSON format.
+        Export the set of courses generically by encoding them in JSON format, plus the HTTP method to use.
         """
-        yield json.dumps(self.courses, sort_keys=True).encode('utf-8')
+        yield json.dumps(self.courses, sort_keys=True).encode('utf-8'), 'POST'
```

Comment:
In order to allow Degreed's course metadata transmitter to just re-use the base integrated channel transmitter's logic (I think SAPSF was already doing so), I had to make the base one use a configurable HTTP method, since Degreed Integration should be able to POST or DELETE depending on the course's enrollability. It should be a good interface to follow for future integrations that will also use methods besides just POST, like SAPSF was.

Reply:
Oh, and the reason it came up now (good thing it did) and not before, is because I noticed I'd have to rewrite almost the same exact test code for Degreed that exists for SAPSF to get full coverage in its course metadata transmitter, which didn't feel right of course.

Reply:
@UmanShahzad Sounds good, thanks for the detailed answers!